In [1]:
import os
import numpy as np
import pandas as pd
from sklearn.metrics import f1_score

### Importing the test set

In [3]:
test = pd.read_csv("../data/test_ne_imputed.csv")
target = 'category'
y_test = test[target].map({"R": 0, "S": 1}).values

### Importing our predictions from various models

In [4]:
temp = pd.read_csv("../subs/temporal_models.csv")
pretrained_emb = pd.read_csv("../subs/pretrained_emb.csv")
frequency_models = pd.read_csv("../subs/frequency_models.csv")
custom_w2v = pd.read_csv("../subs/custom_w2v.csv")

subs = pd.concat([temp, pretrained_emb, frequency_models, custom_w2v], axis=1)
subs.shape

(92, 15)

In [5]:
T = 0.5
ens_1 = (subs['flipcnn_backward'] + subs['avg_3'])/2

print("Final F1 Score : {}".format(f1_score(y_test, (ens_1>T).astype(int))))

Final F1 Score : 0.9541284403669724


In [7]:
sub = pd.DataFrame()
sub['category'] = (ens_1>T).astype(int)
sub['category'] = sub['category'].map({0: "R", 1: "S"})
sub.to_csv("../subs/final_ensemble.csv", index=False)
sub.head()

category
0        R
1        S
2        S
3        S
4        R

# BEST F1 : 0.9541

# Final Ensemble of 4 models:  [FlipCNN_Backward + NB + LR + LGB]